brew_view/controllers/system_list_api.py

```diff
@@ -42,20 +63,48 @@ def get(self):
         tags:
           - Systems
         """
-        self.logger.debug("Getting all Systems")
-
+        query_set = System.objects.order_by(self.request.headers.get('order_by', 'name'))
+        serialize_params = {'to_string': True, 'many': True}
+
+        include_fields = self.request.headers.get('include_fields')
+        exclude_fields = self.request.headers.get('exclude_fields')
+        include_commands = self.get_query_argument('include_commands', None)
+
+        if include_fields and exclude_fields:
+            raise ModelError("Headers 'include_fields' and 'exclude_fields' "
+                             "cannot exist on the same request")
+        elif include_commands and (include_fields or exclude_fields):
+            raise ModelError("Headers 'include_fields' and 'exclude_fields' "
+                             "cannot be used with 'include_commands'")
+        elif include_fields:
+            include_fields = set(include_fields.split(',')) & self.REQUEST_FIELDS
+            query_set = query_set.only(*include_fields)
+            serialize_params['only'] = include_fields
+        elif exclude_fields:
+            exclude_fields = set(exclude_fields.split(',')) & self.REQUEST_FIELDS
+            query_set = query_set.exclude(*exclude_fields)
+            serialize_params['exclude'] = exclude_fields
+
+        # Deal with include_commands
+        if include_commands:
+            if include_commands.lower() == 'false':
+                query_set = query_set.exclude('commands')
+                serialize_params['include_commands'] = False
+            elif include_commands.lower() == 'count':
```

Comment:
I'm not in love with mixing types, but the only other way to do it is to introduce a new query param.

Reply:
I'm not opposed to a new query param - that way we can deprecate `include_commands` and pick a better name.

Reply:
Also, then we can put that param as a header (like include/exclude fields). Then whenever we do a breaking api change we can remove `include_commands` completely and query args will only be used to filter (no more special case we need to check for).